### PR TITLE
Replace docs/reference shortcode with ref URIs

### PR DESCRIPTION
**Pull request by jdbaldry**
You can use `ref` URIs in admonitions (or any shortcodes) because they are inline and not subject to the issues noted in the [`admonition` shortcode](https://grafana.com/docs/writers-toolkit/write/shortcodes/#code-shortcode:~:text=to%20core%20understanding.-,WARNING,For%20more%20information%2C%20refer%20to%20Markdown%20Reference%20Links%20in%20Shortcodes.,-Examples).

The `ref` URIs perform the same pattern matching as `docs/reference` but don't require the use of reference-style links and the destinations are ordinary (full) URLs that can include version substitution. Unlike `docs/reference`, the implementation doesn't use `relref` so you don't have to be careful with omitting trailing slashes and the links will follow redirects.

Documentation: https://grafana.com/docs/writers-toolkit/write/links/#link-from-source-content-thats-used-in-multiple-projects

To check the links, refer to the deploy preview in https://github.com/grafana/website/pull/19630.

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

### CODE REVIEW

**Comment by joeyorlando** [comment]
@jdbaldry I'm guessing @alyssawada would be the go-to person here for a review? 👀

File: docs/sources/manage/insights-and-metrics/index.md [hunk 1]
```diff
@@ -264,7 +270,7 @@ Resource IDs are used a lot in insight logs. You can find them in web ui (exampl
 3. The URL looks like `https://<YOUR_STACK_SLUG>/a/grafana-oncall-app/integrations/C5VXMIFKKP67K`.
 4. Integration ID is `C5VXMIFKKP67K`.
 
-Alternatively you can find the resource ID using the [Grafana OnCall API] or browser dev tools.
+Alternatively you can find the resource ID using the[Grafana OnCall API](ref:grafana-oncall-api) or browser dev tools.
```
Comment (jdbaldry) [inline docs/sources/manage/insights-and-metrics/index.md:273]:
```suggestion
Alternatively you can find the resource ID using the [Grafana OnCall API](ref:grafana-oncall-api) or browser dev tools.
```

File: docs/sources/configure/_index.md [hunk 2]
```diff
@@ -28,7 +44,7 @@ Explore key settings and integrations that allow you to customize your incident
 
 Integrate with your tools and alert sources to begin routing alerts with Grafana OnCall.
 
-For detailed information on configuring integrations, refer to the [Integrations] documentation.
+For detailed information on configuring integrations, refer to the[Integrations](ref:integrations) documentation.
```
Comment (jdbaldry) [inline docs/sources/configure/_index.md:47]:
```suggestion
For detailed information on configuring integrations, refer to the [Integrations](ref:integrations) documentation.
```

File: docs/sources/integrations/_index.md [hunk 3]
```diff
@@ -172,7 +213,7 @@ The Alert Group Labeling feature allows users to:
 - Assign labels to alert groups
 - Filter alert groups by labels
 - Customize the Alert Group table
-- Pass labels in [Webhooks]
+- Pass labels in[Webhooks](ref:webhooks)
```
Comment (jdbaldry) [inline docs/sources/integrations/_index.md:216]:
```suggestion
- Pass labels in [Webhooks](ref:webhooks)
```

File: docs/sources/integrations/inbound-email/index.md [hunk 4]
```diff
@@ -18,7 +29,7 @@ Inbound Email integration will consume emails from dedicated email address and m
 
 ## Configure required environment variables
 
-Refer to [Inbound Email Setup] for details.
+Refer to[Inbound Email Setup](ref:inbound-email-setup) for details.
```
Comment (jdbaldry) [inline docs/sources/integrations/inbound-email/index.md:32]:
```suggestion
Refer to [Inbound Email Setup](ref:inbound-email-setup) for details.
```

**Review by alyssawada — APPROVED** [review]
@jdbaldry thanks for taking the time to update this. LGTM!

**Pull request merged**